shells/browser/shared/src/main.js

```diff
@@ -54,69 +63,85 @@ function createPanelIfReactLoaded() {
+                  attribute.nodeName,
+                  attribute.nodeValue
+                );
+              }
+              newLinkTags.push(newLinkTag);
+            }
+          }
+
+          return newLinkTags;
+        };
+
         if (elementsPanel !== null) {
-          elementsPanel.injectBridgeAndStore(bridge, store);
-        }
-        if (profilerPanel !== null) {
-          profilerPanel.injectBridgeAndStore(bridge, store);
-        }
-        if (settingsPanel !== null) {
-          settingsPanel.injectBridgeAndStore(bridge, store);
+          elementsPanel.render(renderRootToPortal, 'elements');
         }
       }
 
       initBridgeAndStore();
 
-      chrome.devtools.panels.create(
-        '⚛ Elements',
-        '',
-        'elements.html',
-        panel => {
-          panel.onShown.addListener(panel => {
-            if (elementsPanel === null) {
-              panel.injectBridgeAndStore(bridge, store);
-            }
+      chrome.devtools.panels.create('⚛ Elements', '', 'panel.html', panel => {
+        panel.onShown.addListener(panel => {
+          elementsPanel = panel;
 
-            elementsPanel = panel;
+          if (renderRootToPortal !== null) {
+            elementsPanel.render(renderRootToPortal, 'elements');
+          }
 
-            // TODO: When the user switches to the panel, check for an Elements tab selection.
-          });
-          panel.onHidden.addListener(() => {
-            // TODO: Stop highlighting and stuff.
-          });
-        }
-      );
+          // TODO: When the user switches to the panel, check for an Elements tab selection.
+        });
+        panel.onHidden.addListener(() => {
+          // TODO: Stop highlighting and stuff.
+        });
+      });
 
       // TODO (profiling) Is there a way to detect profiling support and conditionally register this panel?
-      chrome.devtools.panels.create(
-        '⚛ Profiler',
-        '',
-        'profiler.html',
-        panel => {
-          panel.onShown.addListener(panel => {
-            if (settingsPanel === null) {
-              panel.injectBridgeAndStore(bridge, store);
-            }
+      chrome.devtools.panels.create('⚛ Profiler', '', 'panel.html', panel => {
+        panel.onShown.addListener(panel => {
+          profilerPanel = panel;
 
-            profilerPanel = panel;
-          });
-        }
-      );
-
-      chrome.devtools.panels.create(
-        '⚛ Settings',
-        '',
-        'settings.html',
-        panel => {
-          panel.onShown.addListener(panel => {
-            if (settingsPanel === null) {
-              panel.injectBridgeAndStore(bridge, store);
-            }
+          if (renderRootToPortal !== null) {
+            profilerPanel.render(renderRootToPortal, 'profiler');
+          }
+        });
+      });
 
-            settingsPanel = panel;
-          });
-        }
-      );
+      chrome.devtools.panels.create('⚛ Settings', '', 'panel.html', panel => {
```

Comment:
Settings may be more efficient use of space as an overlay on top of an existing panel (on any of them) like the DevTools own settings.

Reply:
That would require putting a toggle inside the panel which would take up precious space.

Anyway, feedback is usually appreciated but I'm not looking for UI/UX feedback on this PR. Only posted it in case anyone had suggestions on the shared styles issue, or the general portaling technique.

Reply:
> That would require putting a toggle inside the panel which would take up precious space.

Yes, but this could be a small cog button or even smaller three-vertical-dots menu on the side.

> Anyway, feedback is usually appreciated but I'm not looking for UI/UX feedback on this PR. Only posted it in case anyone had suggestions on the shared styles issue, or the general portaling technique.

Yeah sorry for the distraction. Noticed this piece of code here so decided to report. There's no other way in GitHub to review/leave feedback on code or product decisions other than in a diff/PR or in an issue after the decisions have been finalized.